### PR TITLE
Use reshare with Diaspora like with ActivityPub

### PR DESCRIPTION
**Pull request by annando**
Since Diaspora doesn't support reshares in a way like AP is doing them, it had been a problem to support them equally. Because of some recent database changes it is now possible to simulate this experience. So from a user's perspective one can now reshare and unreshare a post via web or API and it will look and feel like with AP. The difference is that with Diaspora a separate quoted reshare post is generated or deleted in the background.

This code isn't well tested at the moment and I won't be available until Saturday for feedback. But it shouldn't harm the system, so possibly someone wants to try it out.

### CODE REVIEW

**Comment by MrPetovan** [comment]
I'll switch this PR to Draft until you've tested a bit more.

**Comment by HankG** [comment]
I'd love to test this. Is there a server I can create an account against to try it with the various permutations I run into regularly?
* Friendica post reshared so that it goes to AP and D* networks
* AP federated post that is "reshared" which goes to both AP and D* connected users
* D* federated post that is "reshared" which goes to both AP and D* connected users

**Comment by tobiasd** [comment]
I think this is not a bug fix, so it should not be part of the December release...

File: src/Content/Conversation.php [hunk 1]
```diff
@@ -985,7 +985,8 @@ private function addChildren(array $parents, bool $block_authors, string $order,
 
 		$thread_items = Post::selectForUser($uid, array_merge(ItemModel::DISPLAY_FIELDLIST, ['featured', 'contact-uid', 'gravity', 'post-type', 'post-reason']), $condition, $params);
 
-		$items = [];
+		$items       = [];
+		$quoteuriids = [];
```
Comment (MrPetovan) [inline src/Content/Conversation.php]: Please use either `camelCase` or `snake_case` for variable names.
```suggestion
		$quoteUriIds = [];
```

File: src/Navigation/Notifications/Factory/Notification.php [hunk 2]
```diff
@@ -311,7 +311,11 @@ public function getMessageFromNotification(Entity\Notification $Notification): a
 								$msg = $l10n->t('%1$s shared a post');
 							}
 							break;
-					}
+
+						case Post\UserNotification::TYPE_QUOTED:
+							$msg = $l10n->t('%1$s shared your post %2$s');
+							break;
+						}
```
Comment (MrPetovan) [inline src/Navigation/Notifications/Factory/Notification.php]:
```suggestion
					}
```

**Review by MrPetovan — CHANGES_REQUESTED** [review]

File: src/Content/Conversation.php (hunk 1 above)
Comment (annando) [reply src/Content/Conversation.php]: Done

File: src/Navigation/Notifications/Factory/Notification.php (hunk 2 above)
Comment (annando) [reply src/Navigation/Notifications/Factory/Notification.php]: Done

**Review by MrPetovan — APPROVED** [review]
Is this PR done?

**Comment by annando** [comment]
I think so.

**Pull request merged**